app/code/Magento/Catalog/Model/Product/SalabilityChecker.php

```diff
@@ -25,16 +26,24 @@ class SalabilityChecker
      */
     private $storeManager;
 
+    /**
+     * @var StockConfigurationInterface
+     */
+    private $stockConfiguration;
+
     /**
      * @param ProductRepositoryInterface $productRepository
      * @param StoreManagerInterface $storeManager
+     * @param StockConfigurationInterface $stockConfiguration
      */
     public function __construct(
         ProductRepositoryInterface $productRepository,
-        StoreManagerInterface $storeManager
+        StoreManagerInterface $storeManager,
+        StockConfigurationInterface $stockConfiguration
```

Comment:
Please, inject new dependencies according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

Reply:
Thanks for sharing this link, I have updated my code accordingly.